library/macos/FLEViewController.m

```diff
@@ -219,53 +234,11 @@ - (BOOL)launchEngineWithMainPath:(NSURL *)main
                              commandLineArguments:arguments];
 }
 
-- (BOOL)addPlugin:(id<FLEPlugin>)plugin {
-  NSString *channelName = plugin.channel;
-  if (_messageHandlers[channelName] != nil) {
-    NSLog(@"Warning: channel %@ already has an associated plugin", channelName);
-    return NO;
-  }
-
-  plugin.controller = self;
-  // See the note on the |codec| property of FLEPlugin for the reason this defaults to JSON.
-  id<FLEMethodCodec> codec = [FLEJSONMethodCodec sharedInstance];
-  if ([plugin respondsToSelector:@selector(codec)]) {
-    codec = plugin.codec;
-  }
-
-  // TODO: Move the channel into the plugin implementations, matching Flutter's iOS plugin
-  // structure. This is is a temporary shim to add FLEMethodChannel without coupling it to the
-  // breaking API change.
-  FLEMethodChannel *channel = [FLEMethodChannel methodChannelWithName:channelName
-                                                      binaryMessenger:self
-                                                                codec:codec];
-  // This will end up calling back to setMessageHandlerOnChannel:binaryMessageHandler: below, which
-  // retains the handler, so the strong reference to |plugin| here will keep the plugin alive.
-  [channel setMethodCallHandler:^(FLEMethodCall *call, FLEMethodResult result) {
-    [plugin handleMethodCall:call result:result];
-  }];
-
-  return YES;
-}
-
-- (void)invokeMethod:(NSString *)method
-           arguments:(id)arguments
-           onChannel:(NSString *)channelName
-           withCodec:(id<FLEMethodCodec>)codec {
-  // TODO: Move the channel into the plugin implementations, matching Flutter's iOS plugin
-  // structure. This is is a temporary shim to add FLEMethodChannel without coupling it to the
-  // breaking API change.
-  FLEMethodChannel *channel = [FLEMethodChannel methodChannelWithName:channelName
-                                                      binaryMessenger:self
-                                                                codec:codec];
-  [channel invokeMethod:method arguments:arguments];
-}
-
-- (void)invokeMethod:(NSString *)method arguments:(id)arguments onChannel:(NSString *)channel {
-  [self invokeMethod:method
-           arguments:arguments
-           onChannel:channel
-           withCodec:[FLEJSONMethodCodec sharedInstance]];
+- (id<FLEPluginRegistrar>)registrarForPlugin:(NSString *)pluginName {
+  // Currently, the view controller acts as the registrar for all plugins, so the
+  // name is ignored. It is part of the API to reduce churn in the future when
+  // aligning more closely with the Flutter registrar system.
```

Comment:
Out of curiosity, what does the Flutter registrar system do with the name?

Reply:
Registrars are one-per-plugin in Flutter, so the name is used as a map key for finding the correct registrar.